api/serializers.py

```diff
@@ -51,3 +50,23 @@ class Meta:
     def to_representation(self, instance):
         representation = super().to_representation(instance)
         return representation['follower']
+
+
+class LikesSerializer(serializers.ModelSerializer):
+    parent_lookup_kwargs = {
+        'author_pk': 'author__pk',
+    }
+    author = AuthorSerializer(many=False, required=True)
+    type = serializers.CharField(default='Like', read_only=True)
+
+    class Meta:
+        model = Like
+        fields = ['type', 'author', 'summary', 'object']
+
+    def create(self, validated_data):
+        # author_data = validated_data["author"]
+        # if author_data:
+        #     author = get_user_model().objects.get(**author_data)
+        #     validated_data["author"] = author
+        like = Like.objects.get_or_create(**validated_data)
+        return like
```

Comment:
```suggestion
class LikesSerializer(NestedHyperlinkedModelSerializer):
    parent_lookup_kwargs = {
        'author_pk': 'author__pk',
        'post_pk': 'post__pk',
    }
    author = AuthorSerializer(many=False, read_only=True)
    post = PostSerializer(many=False, read_only=True)

    class Meta:
        model = Like
        fields = ['author', 'post']

    def to_representation(self, instance):
        representation = super().to_representation(instance)
        representation['object'] = representation['post']['url']
        del representation['post']
        return representation

```

Reply:
+1 to this serialize suggestion; the nested sterilizers can be really tricky but this is a great solution